### PR TITLE
Photo Gallery with animated transitions and carousel

### PR DESCRIPTION
**Pull request by leealbert95**
Please review my code for this submission

### CODE REVIEW

**Review by sujithavikram — COMMENTED** [review]
@leealbert95  Looks like you covered almost everything in unit test. well done !!

File: client/components/ImageService.jsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    this.fetchNewImages = this.fetchNewImages.bind(this);
+    this.onImageLoad = this.onImageLoad.bind(this);
+    this.openModal = this.openModal.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+    this.changeIndex = this.changeIndex.bind(this);
+  }
+
+  componentDidMount() {
+    console.log(this.props.locationId);
+    this.fetchNewImages(this.props.locationId);
```
Comment (2000prcs) [inline client/components/ImageService.jsx:30]: Don't forget to remove console.log! :)

File: client/components/ImageService.jsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+  onImageLoad() {
+    let {imageCount, allImagesLoaded, images} = this.state;
+    if (imageCount === images.length) {
+      return;
+    }
+    imageCount++;
+    allImagesLoaded = imageCount === images.length;
+    console.log('Images loaded: ' + allImagesLoaded);
+    this.setState({
```
Comment (2000prcs) [inline client/components/ImageService.jsx:47]: Another console.log here!

File: client/icons/left-arrow.svg [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" id="Capa_1" x="0px" y="0px" viewBox="0 0 377.406 377.406" style="enable-background:new 0 0 377.406 377.406;" xml:space="preserve" width="512px" height="512px" class=""><g transform="matrix(1 0 0 1 0 0)"><g>
+	<path d="M270.652,373.981c11.583,10.618,27.993-5.792,17.375-17.375C230.11,300.62,173.159,244.633,115.242,188.647   C173.159,132.661,230.11,76.675,288.027,20.688c11.583-10.618-5.792-27.993-17.375-17.375   C209.839,62.195,149.027,121.078,89.179,179.96c-4.826,4.826-4.826,12.549,0,17.375   C149.027,256.217,209.839,315.099,270.652,373.981z" data-original="#000000" class="active-path" data-old_color="#150B0B" fill="#FFFFFF"/>
+	<g>
+	</g>
+	<g>
+	</g>
```
Comment (2000prcs) [inline client/icons/left-arrow.svg:7]: Why are there so many 'g' tags here?

**Comment by 2000prcs** [comment]
@leealbert95 Overall it looks good! Just left few comments.

File: client/icons/left-arrow.svg (hunk 3 above)
Comment (leealbert95) [reply client/icons/left-arrow.svg:7]: No idea, that's the code I got from the website. In the jsx versions of the icons I cut out all the extra <g> tags and it still looks fine

**Comment by leealbert95** [comment]
Alright, I'm going to merge this now

**Pull request merged**